File: riveredge-backend/src/tree_root/models/message_template.py
"""
消息模板模型模块

定义消息模板数据模型，用于消息管理。
"""

from tortoise import fields
from typing import Optional, Dict, Any
from .base import BaseModel


class MessageTemplate(BaseModel):
    """
    消息模板模型
    
    用于定义和管理组织内的消息模板。
    支持多组织隔离，每个组织的消息模板相互独立。
    
    注意：继承自 BaseModel，自动包含 uuid、tenant_id、created_at、updated_at 字段。
    """
    id = fields.IntField(pk=True, description="消息模板ID（主键，自增ID，内部使用）")
    # uuid 字段由 BaseModel 提供
    # tenant_id 字段由 BaseModel 提供
    
    name = fields.CharField(max_length=100, description="模板名称")
    code = fields.CharField(max_length=50, description="模板代码（唯一，用于程序识别）")
    type = fields.CharField(max_length=20, description="消息类型（email、sms、internal、push）")
    description = fields.TextField(null=True, description="模板描述")
    
    subject = fields.CharField(max_length=200, null=True, description="主题（邮件、推送通知）")
    content = fields.TextField(description="模板内容（支持变量）")
    variables = fields.JSONField(null=True, description="模板变量定义（JSON格式）")
    
    is_active = fields.BooleanField(default=True, description="是否启用")
    
    # 软删除字段
    deleted_at = fields.DatetimeField(null=True, description="删除时间（软删除）")
    
    class Meta:
        """
        模型元数据
        """
        table = "root_message_templates"
        indexes = [
            ("tenant_id", "code"),  # 唯一索引
            ("type",),
            ("uuid",),
            ("code",),
            ("created_at",),
        ]
        unique_together = [("tenant_id", "code")]
    
    def __str__(self):
        """字符串表示"""
        return f"{self.name} ({self.type})"

File: riveredge-frontend/src/services/messageTemplate.ts
/**
 * 消息模板管理服务
 * 
 * 提供消息模板的 CRUD 操作。
 * 注意：所有 API 自动过滤当前组织的消息模板
 */

import { apiRequest } from './api';

export interface MessageTemplate {
  uuid: string;
  tenant_id: number;
  name: string;
  code: string;
  type: 'email' | 'sms' | 'internal' | 'push';
  description?: string;
  subject?: string;
  content: string;
  variables?: Record<string, any>;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface MessageTemplateListParams {
  skip?: number;
  limit?: number;
  type?: string;
  is_active?: boolean;
}

export interface CreateMessageTemplateData {
  name: string;
  code: string;
  type: 'email' | 'sms' | 'internal' | 'push';
  description?: string;
  subject?: string;
  content: string;
  variables?: Record<string, any>;
  is_active?: boolean;
}

export interface UpdateMessageTemplateData {
  name?: string;
  description?: string;
  subject?: string;
  content?: string;
  variables?: Record<string, any>;
  is_active?: boolean;
}

/**
 * 获取消息模板列表
 */
export async function getMessageTemplateList(params?: MessageTemplateListParams): Promise<MessageTemplate[]> {
  return apiRequest<MessageTemplate[]>('/core/message-templates', {
    params,
  });
}

/**
 * 获取消息模板详情
 */
export async function getMessageTemplateByUuid(messageTemplateUuid: string): Promise<MessageTemplate> {
  return apiRequest<MessageTemplate>(`/core/message-templates/${messageTemplateUuid}`);
}

/**
 * 创建消息模板
 */
export async function createMessageTemplate(data: CreateMessageTemplateData): Promise<MessageTemplate> {
  return apiRequest<MessageTemplate>('/core/message-templates', {
    method: 'POST',
    data,
  });
}

/**
 * 更新消息模板
 */
export async function updateMessageTemplate(messageTemplateUuid: string, data: UpdateMessageTemplateData): Promise<MessageTemplate> {
  return apiRequest<MessageTemplate>(`/core/message-templates/${messageTemplateUuid}`, {
    method: 'PUT',
    data,
  });
}

/**
 * 删除消息模板
 */
export async function deleteMessageTemplate(messageTemplateUuid: string): Promise<void> {
  return apiRequest<void>(`/core/message-templates/${messageTemplateUuid}`, {
    method: 'DELETE',
  });
}

